plugin/path_checkout_sets.go

```diff
@@ -114,41 +128,36 @@ func (b *backend) operationReserveCreate(ctx context.Context, req *logical.Reque
 		return logical.ErrorResponse(`"service_account_names" must be provided`), nil
 	}
 	// Ensure these service accounts aren't already managed by another check-out set.
-	var alreadyManagedErr error
+	var userErrs error
+	var resultingServiceAccountNames []string
 	for _, serviceAccountName := range serviceAccountNames {
-		if _, err := b.checkOutHandler.Status(ctx, req.Storage, serviceAccountName); err != nil {
-			if err == ErrNotFound {
-				// This is what we want to see.
+		if isUserErr, err := b.checkInNewServiceAccount(ctx, req.Storage, serviceAccountName); err != nil {
+			if isUserErr {
+				userErrs = multierror.Append(userErrs, err)
 				continue
 			}
-			// There is a more persistent error reaching storage.
-			return nil, err
-		}
-		// If we reach here, the error is nil. That means there's an existing CheckOut for this
-		// service account.
-		alreadyManagedErr = multierror.Append(alreadyManagedErr, fmt.Errorf("%s is already managed by another set, please remove it and try again", serviceAccountName))
-	}
-	if alreadyManagedErr != nil {
-		return logical.ErrorResponse(alreadyManagedErr.Error()), nil
-	}
-
-	// Now we need to check in all these service accounts so they'll be listed as managed by this
-	// plugin and available.
-	for _, serviceAccountName := range serviceAccountNames {
-		if err := b.checkOutHandler.CheckIn(ctx, req.Storage, serviceAccountName); err != nil {
 			return nil, err
 		}
+		resultingServiceAccountNames = append(resultingServiceAccountNames, serviceAccountName)
 	}
 
 	set := &librarySet{
-		ServiceAccountNames:       serviceAccountNames,
+		ServiceAccountNames:       resultingServiceAccountNames,
 		TTL:                       ttl,
 		MaxTTL:                    maxTTL,
 		DisableCheckInEnforcement: disableCheckInEnforcement,
 	}
+	if err := set.Validate(); err != nil {
```

Comment:
Shouldn't this check happen before we've already checked-in the new service accounts?

Reply:
Ugh, that's kind of a catch-22. If we're taking an approach where we check in the ones we can, then I have to wait until we're done with that to validate the set, because I won't know which service accounts we're using until we're done.

Reply:
Fixed in latest refactor, much easier without partial updates.